integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractStreamIndexingTest.java

```diff
@@ -0,0 +1,439 @@
+  static final DateTimeFormatter TIMESTAMP_FMT = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss'.000Z'");
+  static final int EVENTS_PER_SECOND = 6;
+  static final int TOTAL_NUMBER_OF_SECOND = 10;
+  static final Logger LOG = new Logger(AbstractStreamIndexingTest.class);
+  // Since this integration test can terminates or be killed un-expectedly, this tag is added to all streams created
+  // to help make stream clean up easier. (Normally, streams should be cleanup automattically by the teardown method)
+  // The value to this tag is a timestamp that can be used by a lambda function to remove unused stream.
+  private static final String STREAM_EXPIRE_TAG = "druid-ci-expire-after";
+  private static final int STREAM_SHARD_COUNT = 2;
+  private static final long WAIT_TIME_MILLIS = 3 * 60 * 1000L;
+  private static final String INDEXER_FILE_LEGACY_PARSER = "/indexer/stream_supervisor_spec_legacy_parser.json";
+  private static final String INDEXER_FILE_INPUT_FORMAT = "/indexer/stream_supervisor_spec_input_format.json";
+  private static final String QUERIES_FILE = "/indexer/stream_index_queries.json";
+  private static final long CYCLE_PADDING_MS = 100;
+
+  @Inject
+  private DruidClusterAdminClient druidClusterAdminClient;
+
+  private StreamAdminClient streamAdminClient;
+  private WikipediaStreamEventStreamGenerator wikipediaStreamEventGenerator;
+
+  abstract StreamAdminClient getStreamAdminClient() throws Exception;
+  abstract StreamEventWriter getStreamEventWriter() throws Exception;
+  abstract Function<String, String> generateStreamIngestionPropsTransform(String streamName, String fullDatasourceName);
+  abstract Function<String, String> generateStreamQueryPropsTransform(String streamName, String fullDatasourceName);
+  public abstract String getTestNamePrefix();
+
+  protected void doBeforeClass() throws Exception
+  {
+    streamAdminClient = getStreamAdminClient();
+    wikipediaStreamEventGenerator = new WikipediaStreamEventStreamGenerator(EVENTS_PER_SECOND, CYCLE_PADDING_MS);
+  }
+
+  protected void doClassTeardown()
+  {
+    wikipediaStreamEventGenerator.shutdown();
+  }
+
+  protected void doTestIndexDataWithLegacyParserStableState() throws Exception
+  {
+    StreamEventWriter streamEventWriter = getStreamEventWriter();
+    final GeneratedTestConfig generatedTestConfig = new GeneratedTestConfig();
+    try (
+        final Closeable ignored1 = unloader(generatedTestConfig.getFullDatasourceName())
+    ) {
+      final String taskSpec = generatedTestConfig.getStreamIngestionPropsTransform().apply(getResourceAsString(INDEXER_FILE_LEGACY_PARSER));
+      LOG.info("supervisorSpec: [%s]\n", taskSpec);
+      // Start supervisor
+      generatedTestConfig.setSupervisorId(indexer.submitSupervisor(taskSpec));
+      LOG.info("Submitted supervisor");
+      // Start data generator
+      wikipediaStreamEventGenerator.start(generatedTestConfig.getStreamName(), streamEventWriter, TOTAL_NUMBER_OF_SECOND, FIRST_EVENT_TIME);
```

Comment:
Probably better to call `run()` or `generate()` rather than `start()` since the generator is not something you can start/stop, but generates all events in `start()`.

Reply:
Done